File: go-client-sdk/pkg/models/operations/getcompaniescompanyidfilesdownload.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/pkg/models/shared"
	"time"
)

type GetCompaniesCompanyIDFilesDownloadPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
}

type GetCompaniesCompanyIDFilesDownloadQueryParams struct {
	Date *time.Time `queryParam:"style=form,explode=true,name=date"`
}

type GetCompaniesCompanyIDFilesDownloadSecurity struct {
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
}

type GetCompaniesCompanyIDFilesDownloadRequest struct {
	PathParams  GetCompaniesCompanyIDFilesDownloadPathParams
	QueryParams GetCompaniesCompanyIDFilesDownloadQueryParams
	Security    GetCompaniesCompanyIDFilesDownloadSecurity
}

type GetCompaniesCompanyIDFilesDownloadResponse struct {
	ContentType string
	StatusCode  int64
}

File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyidfilesdownload.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class GetCompaniesCompanyIdFilesDownloadPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;
}

export class GetCompaniesCompanyIdFilesDownloadQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=date" })
  date?: Date;
}

export class GetCompaniesCompanyIdFilesDownloadSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetCompaniesCompanyIdFilesDownloadRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdFilesDownloadPathParams;

  @SpeakeasyMetadata()
  queryParams: GetCompaniesCompanyIdFilesDownloadQueryParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdFilesDownloadSecurity;
}

export class GetCompaniesCompanyIdFilesDownloadResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}